modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/navigate.es.js

```diff
@@ -32,7 +32,25 @@ export default function (url, listeners) {
 			});
 		}
 	}
-	else {
+	else if (_isValidURL(url)) {
 		window.location.href = url;
 	}
 }
+
+function _isValidURL(url) {
+	var url_object;
+
+	try {
+		if (url.startsWith('/')) {
+			url_object = new URL(url, window.location.origin);
+		}
+		else {
+			url_object = new URL(url);
+		}
+	}
+	catch (error) {
+		return false;
+	}
+
+	return url_object.protocol === 'http:' || url_object.protocol === 'https:';
```

Comment:
Nice touch with the protocol check! :+1:

Reply:
Thank you for your review and feedback!